aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/BaseHandlerStd.java

```diff
@@ -30,4 +44,83 @@
     final CallbackContext callbackContext,
     final ProxyClient<CodeartifactClient> proxyClient,
     final Logger logger);
+
+  protected ProgressEvent<ResourceModel, CallbackContext> associateExternalConnections(
+      final ProgressEvent<ResourceModel, CallbackContext> progress,
+      final CallbackContext callbackContext,
+      final ResourceHandlerRequest<ResourceModel> request,
+      final ProxyClient<CodeartifactClient> proxyClient,
+      final Set<String> externalConnectionsToAdd,
+      final Logger logger
+  ) {
+      ResourceModel resourceModel = request.getDesiredResourceState();
+
+      if (CollectionUtils.isNullOrEmpty(externalConnectionsToAdd)) {
+          // Nothing to add, continue
+          return ProgressEvent.progress(resourceModel, callbackContext);
+      }
+
+      if (externalConnectionsToAdd.size() > 1) {
+          return ProgressEvent.failed(resourceModel, callbackContext, HandlerErrorCode.InvalidRequest,
+              "ExternalConnections for repositories are currently limited to 1.");
+      }
+
+      // Loop is currently not necessary because only 1 external connection is allowed, leaving this in for
+      // when multiple are supported
+      externalConnectionsToAdd.forEach(ec -> {
+          try {
+              AssociateExternalConnectionRequest associateExternalConnectionRequest
+                  = Translator.translateAssociateExternalConnectionsRequest(resourceModel, ec);
+
+              proxyClient.injectCredentialsAndInvokeV2(associateExternalConnectionRequest, proxyClient.client()::associateExternalConnection);
+          } catch (final AwsServiceException e) {
+              String repositoryName = progress.getResourceModel().getRepositoryName();
+              Translator.throwCfnException(e, Constants.ASSOCIATE_EXTERNAL_CONNECTION, repositoryName);
+          }
+          logger.log(String.format("Successfully associated external connection: %s", ec));
+      });
+
```

Comment:
What I did to get around this is to disassociate before associating new ECs so that if there is one attached which is different than the one requested, it will dissociate and then associate the new one:

https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeartifact/pull/20/files#diff-fced23059cfaaacbec9831b63127cb21R50